anvil/src/eth/api.rs

```diff
@@ -71,6 +73,62 @@ use super::backend::mem::BlockRequest;
 /// The client version: `anvil/v{major}.{minor}.{patch}`
 pub const CLIENT_VERSION: &str = concat!("anvil/v", env!("CARGO_PKG_VERSION"));
 
+#[derive(Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NodeInfo {
+    current_block_number: U64,
+    current_block_timestamp: u64,
+    current_block_hash: H256,
+    hard_fork: SpecId,
+    transaction_order: String,
+    environment: NodeEnvironment,
+    fork_config: NodeForkConfig,
+}
+
+#[derive(Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NodeEnvironment {
+    base_fee: U256,
+    chain_id: U256,
+    gas_limit: U256,
+    gas_price: U256,
+}
+
+#[derive(Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NodeForkConfig {
+    fork_url: Option<String>,
+    fork_block_number: Option<u64>,
+    fork_retry_backoff: Option<u128>,
+}
+
+impl NodeInfo {
+    pub fn new(
+        current_block_number: U64,
+        current_block_timestamp: u64,
+        current_block_hash: H256,
+        hard_fork: SpecId,
+        transaction_order: String,
+        base_fee: U256,
+        chain_id: U256,
+        gas_limit: U256,
+        gas_price: U256,
+        fork_url: Option<String>,
+        fork_block_number: Option<u64>,
+        fork_retry_backoff: Option<u128>,
+    ) -> NodeInfo {
```

Comment:
since we have a lot of fields here, I suggest we ditch the `new` function, make all fields `pub` and initialize directly

Reply:
done in https://github.com/foundry-rs/foundry/pull/3779/commits/669c0dd6c9cc9725825824aef0f0c1be1bf65405